bpo-35904: Add statistics.fmean()

rhettinger opened the pull request:
<!-- issue-number: [bpo-35904](https://bugs.python.org/issue35904) -->
https://bugs.python.org/issue35904
<!-- /issue-number -->

rhettinger committed:
Add statistics.fmean()

scoder commented on Lib/statistics.py:
```diff
@@ -312,6 +312,33 @@ def mean(data):
+    try:
+        return total / n
+    except ZeroDivisionError:
+        raise StatisticsError('fmean requires at least one data point')
```
Is it really helpful to keep the exception context of the `ZeroDivisionError` here? I'm just asking, can live with it either way.

rhettinger replied:
The error message is clear about about the cause, so the exception context isn't helpful.  I'll make the change.  Thanks for noticing.

scoder reviewed (approved):
Patch looks very good to me.

rhettinger committed:
Suppress exception chaining because it doesn't add value here.

rhettinger committed:
Add an attribution

rhettinger committed:
Replace mean() is fmean() in resampling and bootstrapping examples.

In both examples, the mean() is called in a loop and the results
are expected to be floats.  In the context of Monte Carlo
estimation, high accuracy rather than perfect accuracy will suffice.
